app/Http/Controllers/Mship/WaitingLists.php

```diff
@@ -47,4 +48,24 @@ public function selfEnrol(WaitingList $waitingList, Request $request)
             ->route('mship.waiting-lists.index')
             ->with('success', 'You have been added to the waiting list.');
     }
+
+    public function getRetentionWithToken()
+    {
+        $token = request()->query('token');
+
+        if (! $token) {
+            return abort(404, 'No code provided');
+        }
+
+        $retentionCheck = WaitingListRetentionChecks::where('token', $token)->first();
+
+        // Only the scheduled command will change the status so we need to check the expires_at timestamp as well
+        if ($retentionCheck == null || $retentionCheck->status !== WaitingListRetentionChecks::STATUS_PENDING || $retentionCheck->expires_at < now()) {
+            return abort(404, 'Invalid or expired code');
+        }
+
+        $retentionCheck->response_at = now();
+        $retentionCheck->status = WaitingListRetentionChecks::STATUS_USED;
+        $retentionCheck->save();
```

Comment:
What sort of success do we show to the user once they have clicked on the token? They might not be logged in at the time but we'd benefit from some form of flash message (not sure I have the solution here since that does require auth), or an email confirmation to say they have had their check saved. Otherwise they have no feedback